addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -120,9 +120,17 @@ func (r *VSphereCPIConfigReconciler) mapCPIConfigToDataValuesNonParavirtual( //
 
 	d.Region = tryParseString(d.Region, c.Region)
 	d.Zone = tryParseString(d.Zone, c.Zone)
+
+	d.InsecureFlag = d.TLSThumbprint == ""
 	if c.Insecure != nil {
 		d.InsecureFlag = *c.Insecure
 	}
+	if !d.InsecureFlag && d.TLSThumbprint == "" {
+		return nil, errors.New("empty thumbprint is provided while TLS peer verification is enabled")
+	}
+	if d.InsecureFlag && d.TLSThumbprint != "" {
```

Comment:
we do need to at least log it as warning. 

the value of `d.InsecureFlag` can be optionally specified and override by the Config user in `d.InsecureFlag = *c.Insecure`.

so it's possible that insecure flag is on while having an non-empty thumbprint.

Reply:
That's what I thought you might mean. 
but if you do this: 
```
f d.InsecureFlag {
         	r.Log.Info("ignore thumbprint while TLS peer verification is disabled")
	}
```
The use _will_ get the error if they specify tlsthumbprint and d.insecureFlag=True. 

i.e. if the user sets tlsthumbprint="somevalue" but d.insecureFlag=True, they will see the error above. 

But if you really want to specifically say that the user has provided a tlsthumbprint but it will be ignore because inseureFlag = true then change the wording of the error, because right now it does not read that way. 

small changes but perhaps make it: 
" ignoring provided tlsthumbprint because TLS peer verification is disabled"

Reply:
sure, changed the error message to 

```
		r.Log.Info("ignore provided thumbprint because TLS peer verification is disabled", "thumbprint", d.TLSThumbprint)

```